ktor-client/ktor-client-features/ktor-client-json/common/src/io/ktor/client/features/json/JsonFeature.kt

```diff
@@ -79,13 +81,21 @@ class JsonFeature internal constructor(
             get() = _acceptContentTypes
 
         /**
-         * Adds accepted content types. Be aware that [ContentType.Application.Json] accepted by default is removed from
-         * the list if you use this function to provide accepted content types.
-         * It also affects `Accept` request header value.
+         * Adds accepted content types.
          */
         fun accept(vararg contentTypes: ContentType) {
             _acceptContentTypes += contentTypes
         }
+
+        /**
+         * Defines when to handle a given [ContentType] (used in addition to [accept]).
+         *
+         * By default accepts any `application/<...>+json` pattern.
+         */
+        var shouldHandle: (ContentType) -> Boolean = {
+            val value = it.toString()
+            value.startsWith("application/") && value.endsWith("+json")
```

Comment:
Consider using the `ContentType` class and `match` method. I also would keep the old default, but if you have any reasons we can discuss it

Reply:
The `ContentType` class doesn't support matching on arbitrary patterns like `application/*+json`. The HTTP spec doesn't seem to allow that, either, so I think `ContentType` shouldn't be extended for this use-case.

The only alternatives for matching rules are regexes and lambda functions. I think functions are the most flexible and can still use a regex under the hood very easily.

Reply:
By the way, this enables both "application/json" and "application/...+json" by default. Both content types are very common. That's why I think this default would make sense.

Reply:
Could we fit it in the `ContentType.match` method?

Reply:
The ContentTypeMatcher interface allows this now.